Remove deprecations in DirectoryBuildCache (#33042)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

This property no longer has any effect.
To control cache cleanup behavior in Gradle 9.0 and later, use an <<directory_layout.adoc#dir:gradle_user_home:configure_cache_cleanup,init script>> instead.

==== Removal of `buildCache.local.removeUnusedEntriesAfterDays` property

> In Gradle 9.0, the link:{javadocPath}/org/gradle/caching/local/DirectoryBuildCache.html#setRemoveUnusedEntriesAfterDays-int-[buildCache.local.removeUnusedEntriesAfterDays] property has been removed.

Reviewer: ```suggestion
In Gradle 9.0, the property link:{javadocPath}/org/gradle/caching/local/DirectoryBuildCache.html#setRemoveUnusedEntriesAfterDays-int-[`buildCache.local.removeUnusedEntriesAfterDays`] has been removed.
```

> This property was previously used to configure build cache retention period directly on local build cache.

Reviewer: ```suggestion
This property was previously used to configure the retention period for the local build cache.
```

> To set local build cache retention period, you should now use <<directory_layout.adoc#dir:gradle_user_home:configure_cache_cleanup,Gradle User Home cache cleanup and retention periods configuration>> instead.

Reviewer: ```suggestion
To configure retention for unused entries in the local build cache, use the <<directory_layout.adoc#dir:gradle_user_home:configure_cache_cleanup,Gradle User Home cache cleanup settings>> instead.
```
